modThread/modThread.py

```diff
@@ -16,268 +25,388 @@
 
 
 class ModThread(commands.Cog):
-    """Used to move modmail channels to the correct category for processing by the right team."""
+    """Used to move modmail channels to the correct category for
+    processing by the right team."""
 
     def __init__(self, bot):
         self.bot = bot
         self.config = Config.get_conf(
-            self, identifier=12345678921337, force_registration=True
+            self, identifier=12345678941337, force_registration=True
         )
-        self.config.register_guild(**defaults)
+        self.config.register_guild(**settings)
 
-    # region commands
+# region commands
     @commands.guild_only()
     @commands.command()
-    @checks.admin_or_permissions(manage_guild=True)
-    async def assign(self, ctx, role: str):
+    async def assign(self, ctx, group: str):
         """Assigns the current channel to role and moves channel"""
         currentCategory = ctx.channel.category
-        primaryCategory = await self._primary_category(ctx)
-        rulesCategory = await self._rules_category(ctx)
-        numbersCategory = await self._numbers_category(ctx)
-        modsCategory = await self._mods_category(ctx)
+        primary_category = await self._get_primary_category(ctx.guild)
+        groups = await self._get_groups(ctx.guild)
 
         isThread = False
-        if currentCategory.id == primaryCategory.id:
-            isThread = True
-        elif currentCategory.id == rulesCategory.id:
-            isThread = True
-        elif currentCategory.id == numbersCategory.id:
-            isThread = True
-        elif currentCategory.id == modsCategory.id:
-            isThread = True
-
-        if isThread == False:
+        for group_name in groups:
+            if currentCategory.id == groups[group_name]["category"]:
+                isThread = True
+            elif currentCategory.id == primary_category.id:
```

Comment:
`_get_primary_category()` returns Optional value. If this is NoneType, we generate an exception.